packages/flutter/lib/src/material/ink_well.dart

```diff
@@ -409,47 +455,110 @@ class InkResponse extends StatefulWidget {
   }
 }
 
+/// Used to index the allocated highlights for the different types of highlights
+/// in [_InkResponseState].
+enum _HighlightType {
+  pressed,
+  hover,
+  focus,
+}
+
 class _InkResponseState<T extends InkResponse> extends State<T> with AutomaticKeepAliveClientMixin<T> {
   Set<InteractiveInkFeature> _splashes;
   InteractiveInkFeature _currentSplash;
-  InkHighlight _lastHighlight;
+  FocusNode _focusNode;
+  final Map<_HighlightType, InkHighlight> _highlights = <_HighlightType, InkHighlight>{};
+
```

Comment:
Yes.  You can have a hover, a focus, and a highlight at the same time. (Click on a button that is focused, for instance, since the mouse is over the button, it's got focus, and it's being pressed), and they can't be the same highlight and just change colors, since they are also all animated.